HPXMLtoOpenStudio/resources/electric_panel.rb

```diff
@@ -0,0 +1,237 @@
+  # @param electric_panel [HPXML::ElectricPanel] Object that defines a single electric panel
+  # @param panel_loads [Array<HPXML::PanelLoad>] List of panel load objects
+  # @return [nil]
+  def self.calculate_load_based(hpxml_bldg, electric_panel, panel_loads, panel_calculation_type)
+    if panel_calculation_type == HPXML::ElectricPanelLoadCalculationType2023LoadBased
+      htg_existing = get_panel_load_heating(hpxml_bldg, electric_panel, addition: false)
+      htg_new = get_panel_load_heating(hpxml_bldg, electric_panel, addition: true)
+      clg_existing = electric_panel.panel_loads.select { |panel_load| panel_load.type == HPXML::ElectricPanelLoadTypeCooling && !panel_load.addition }.map { |pl| pl.power }.sum(0.0)
+      clg_new = electric_panel.panel_loads.select { |panel_load| panel_load.type == HPXML::ElectricPanelLoadTypeCooling && panel_load.addition }.map { |pl| pl.power }.sum(0.0)
+
+      # Part A
+      other_load = [htg_existing, clg_existing].max
+      electric_panel.panel_loads.each do |panel_load|
+        next if panel_load.type == HPXML::ElectricPanelLoadTypeHeating || panel_load.type == HPXML::ElectricPanelLoadTypeCooling
+
+        other_load += panel_load.power
+      end
+
+      threshold = 8000.0 # W
+
+      # Part A
+      part_a = 1.0 * [threshold, other_load].min + 0.4 * [0, other_load - threshold].max
+
+      # Part B
+      part_b = [htg_new, clg_new].max
+
+      panel_loads.LoadBased_CapacityW = part_a + part_b
+      panel_loads.LoadBased_CapacityA = panel_loads.LoadBased_CapacityW / Float(electric_panel.voltage)
+      panel_loads.LoadBased_HeadRoomA = electric_panel.max_current_rating - panel_loads.LoadBased_CapacityA
+    elsif panel_calculation_type == HPXML::ElectricPanelLoadCalculationType2026LoadBased
+      # TODO
+      panel_loads.LoadBased_CapacityW = 1
+      panel_loads.LoadBased_CapacityA = 2
+      panel_loads.LoadBased_HeadRoomA = 3
+    end
+  end
+
+  # Calculate the meter-based capacity and headroom for the given electric panel and panel loads according to NEC 220.87.
+  #
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param electric_panel [HPXML::ElectricPanel] Object that defines a single electric panel
+  # @param peak_fuels [Hash] Map of peak building electricity outputs
+  # @return [Array<Double, Double, Double>] The capacity (W), the capacity (A), and headroom (A)
+  def self.calculate_meter_based(hpxml_bldg, electric_panel, peak_fuels, panel_calculation_type)
+    if panel_calculation_type == HPXML::ElectricPanelLoadCalculationType2023MeterBased
+      htg_new = get_panel_load_heating(hpxml_bldg, electric_panel, addition: true)
+      clg_new = electric_panel.panel_loads.select { |panel_load| panel_load.type == HPXML::ElectricPanelLoadTypeCooling && panel_load.addition }.map { |pl| pl.power }.sum(0.0)
+
+      new_loads = [htg_new, clg_new].max
```

Comment:
@lixiliu - I think the latter (i.e., max(CAC, ER furnace)) makes the most sense to me, but you understand the code more deeply than me. Would most interpreters literally only include the added CAC, or would all of "HVAC" be considered new, so you'd take the max between heating and cooling?

Reply:
I think the code is leaving a lot of room for interpretation. After some careful consideration I think we should stick with the code interpretation (i.e., the former). The argument for 2026 revision is the 2023 code is ambiguous and does not allow credits for loads taken out and it is fixing that.

In any case, it isn't max(CAC, ER furnace) because ER furnace is an existing load that makes up the metered peak in the example.

If we start allowing max(0, CAC-ER furnace) as the new load, I fear we are opening doors to allowing loads to be subtracted from the metered peak. Because in the case of a home that already has an ER furnace AND CAC and it is being converted to a HP, we are supposed to add the HP load directly as the new load in 220.87.

I'd vote not to change the logic here.